### ISSUE TITLE
[1.50.3] IFTTT: Test abnormalities and other (enhancements) points

### ISSUE THREAD

**Issue by sfahrenholz** [wekan/wekan#1912]
## Issue

**Server Setup Information**:

* Wekan version: 1.50.3
* Operating System: Ubuntu
* Deployment Method: snap
* Node Version: 8.12.0
* MongoDB Version: from snap

### Other Points as overview

Another points:
* maybe to make rules editable? (see point three in the other lists, if an mistake in the lists name, need to delete the completly rule)

Rules
* when add a checklist, also be possible to add checklist items to the checklist
* make an drop down for the existing lists in a board

_Note: this is only as points, when wanted i create a new "issue"_

Currently fixed after #1936 
Another points:
* ~in the rule dialog a function to go back to the "main menu"~
* ~[Enhancement: Not overwrite first checklist value](https://github.com/wekan/wekan/issues/1912#issuecomment-423827382)~

Translation issue:
- ~the value "r-items-check" exists not in the language file (third option in dialog, maybe "r-item-check"?)~
- ~all values are in lowercase in the "View Rules Dialog"~

Rules
* ~not allow to add an empty rule _(i think related to (1) and (2))_~
* ~[Different Rules with same result give exception](https://github.com/wekan/wekan/issues/1912#issuecomment-423826559)~
* ~[(1) Part 1: Rule "Add Checkliste" not working (Video)](https://github.com/wekan/wekan/issues/1912#issuecomment-423828272)~
* ~[(2) Part 2: Rule "Add Checkliste" not working (issue description old and new)](https://github.com/wekan/wekan/issues/1912#issuecomment-424078901)~

**Comment by xet7** [comment]
@sfahrenholz Thanks for testing edge version!

@Angtrim can look at this.

More info about Wekan Snap stable and edge versions here:
https://github.com/wekan/wekan-snap/wiki/Snap-Developer-Docs

**Comment by Angtrim** [comment]
Hi @sfahrenholz .
I can't reproduce this issue.
Do you have other active rules?
Maybe one that has the same "card moved" trigger?

**Comment by sfahrenholz** [comment]
hey @Angtrim 

i try to reproduce the issue too, but i dont get the error... i playing a little bit more for more details

But i get another issue:
(1) when card moved to list {} add checklist {}
(2) when card added to list {} add checklist {}

then i can reproduce the issue. the added card get no checklist.
![image](https://user-images.githubusercontent.com/6977657/45930021-0a907000-bf5a-11e8-83b2-ef22edcea8d1.png)

**Comment by sfahrenholz** [comment]
I dont know, if this handling is correct or not.. for me as user i didnt expected that
(1) when a card is added to the board add label yellow
(2) when a card is moved to {} add checklist ToDo
(3) when a card is moved to {} add checklist Hallo

In the activities both checklists are added, but only "Hallo" is available, The checklist "ToDo" is override

**Comment by sfahrenholz** [comment]
@Angtrim
Add a Rule with the name "Test"
(1) when card moved to list {} add checklist {}
Add a Rule with the name "Hallo"
(2) when card added to list {} add label YELLOW
Add a Rule with the name "" (empty)
(3) when card moved to list {} add checklist {}

![image](https://user-images.githubusercontent.com/6977657/45971326-13a43e80-c039-11e8-8116-fa0e7f6236ce.png)

I think there is an rule with i added before with an empty name.

**Comment by sfahrenholz** [comment]
Another points:
* in the rule dialog a function to go back to the "main menu"

Rules
* when add a checklist, also be possible to add checklist items to the checklist

**Comment by sfahrenholz** [comment]
Translation issue:

- the value "r-items-check" exists not in the language file (third option in dialog)
- all values are in lowercase in the "View Rules Dialog"

**Comment by sfahrenholz** [comment]
@xet7 @Angtrim 
i cleanup this issue to make a collecting point. i hope this is ok for both of you.

**Comment by sfahrenholz** [comment]
@Angtrim 
Did you need any information about one of the issues? or is anything ambiguous for one of this points?
Let me know if i can help you :)

**Comment by Angtrim** [comment]
@sfahrenholz Everything seems clear, thank you.
I'll proceed fixing those issues asap.

**Comment by therampagerado** [comment]
Is it production ready? Can we merge in develop?

**Comment by danielfaulknor** [comment]
This might be a silly question but I can't work out how to repeat a card, which was one of the most requested features relating to this?

**Comment by xet7** [comment]
@therampagerado 

No, this is not production ready.

-Those above mentioned small issues are not fixed yet.
- I have not yet fixed lint warnings.
- There is no documentation about this in Wekan GitHub wiki yet. If anyone has time to try every possible feature of IFTTT and add documentation to wiki and English screenshots as attachments to this issue in .zip file, it would speed up getting IFTTT production ready.
- I have not had time to even try does every feature of this works. I would need a list about what features IFTTT currently has, and does something not work, as comment to this issue.

**Comment by sfahrenholz** [comment]
@xet7 
documentation as How to used it and how works every rule combination?

I can do the documentation as a how to, when its wanted. When it possible a list of things what is needed were nice.

**Comment by xet7** [comment]
@Angtrim 

About these [lint warnings](https://github.com/wekan/wekan/wiki/Developer-Documentation#preventing-travis-ci-lint-errors-before-submitting-pull-requests) below: do you use these variables at all? There is variables like event, modifier, etc used in fuctions, but they are not used at all in functions. Can they be safely removed? If yes, can you add PR to edge branch?

```
~/repos/wekan$ npm run lint

> wekan@1.52.1 lint /home/user/repos/wekan
> eslint --ignore-pattern 'packages/*' .


/home/user/repos/wekan/client/components/rules/rulesActions.js
  44:38  warning  'event' is defined but never used  no-unused-vars
  47:37  warning  'event' is defined but never used  no-unused-vars
  50:37  warning  'event' is defined but never used  no-unused-vars
  53:42  warning  'event' is defined but never used  no-unused-vars

/home/user/repos/wekan/client/components/rules/rulesMain.js
  27:32  warning  'event' is defined but never used  no-unused-vars

/home/user/repos/wekan/client/components/rules/rulesTriggers.js
  42:39  warning  'event' is defined but never used  no-unused-vars
  45:38  warning  'event' is defined but never used  no-unused-vars
  48:43  warning  'event' is defined but never used  no-unused-vars

/home/user/repos/wekan/models/checklistItems.js
  121:60  warning  'modifier' is defined but never used  no-unused-vars
  139:62  warning  'modifier' is defined but never used  no-unused-vars

/home/user/repos/wekan/models/wekanCreator.js
  513:37  warning  'ruleIndex' is defined but never used  no-unused-vars
  528:35  warning  'ruleIndex' is defined but never used  no-unused-vars
  543:31  warning  'ruleIndex' is defined but never used  no-unused-vars

/home/user/repos/wekan/server/rulesHelper.js
  63:13  warning  'emailTo' is assigned a value but never used       no-unused-vars
  64:13  warning  'emailMsg' is assigned a value but never used      no-unused-vars
  65:13  warning  'emailSubject' is assigned a value but never used  no-unused-vars

✖ 16 problems (0 errors, 16 warnings)
```

**Comment by sfahrenholz** [comment]
@xet7 
for the most important mistakes the fixes are merged. The last three points are enhancements. for me the primary functions are ok - maybe create new "issues" for the enhancements?. So when you released a new edge-version i can testing the functions and create a documentation for using.

**Comment by xet7** [comment]
Wekan v1.53.2 snap+docker included latest IFTTT+LDAP, please test what works, but make sure you have backups. This is very experimental, brand new, and can contain bugs, or not work at all.

**Comment by xet7** [comment]
Moved to #1160
